### PR TITLE
Centralise logging behind JUL

### PR DESCRIPTION
**Pull request by yschimke**
Use JUL logging consistently, and wire in sensible Android behaviour by default.

### CODE REVIEW

**Comment by yschimke** [comment]
Good explanation of log filtering in android here https://mobiarch.wordpress.com/2012/05/11/proper-logging-from-android-application/

**Comment by yschimke** [comment]
Tested with

```
$ adb shell setprop log.tag.okhttp.Http2 DEBUG
```

Output 

```
2020-04-30 08:19:27.253 27576-27609/okhttp.android.test.test D/okhttp.Http2: >> CONNECTION 505249202a20485454502f322e300d0a0d0a534d0d0a0d0a
2020-04-30 08:19:27.255 27576-27609/okhttp.android.test.test D/okhttp.Http2: >> 0x00000000     6 SETTINGS      
2020-04-30 08:19:27.257 27576-27609/okhttp.android.test.test D/okhttp.Http2: >> 0x00000000     4 WINDOW_UPDATE 
2020-04-30 08:19:27.267 27576-27612/okhttp.android.test.test D/okhttp.Http2: << 0x00000000    18 SETTINGS      
2020-04-30 08:19:27.269 27576-27612/okhttp.android.test.test D/okhttp.Http2: << 0x00000000     4 WINDOW_UPDATE 
2020-04-30 08:19:27.282 27576-27609/okhttp.android.test.test D/okhttp.Http2: >> 0x00000003    45 HEADERS       END_STREAM|END_HEADERS
2020-04-30 08:19:27.285 27576-27613/okhttp.android.test.test D/okhttp.Http2: >> 0x00000000     0 SETTINGS      ACK
2020-04-30 08:19:27.300 27576-27612/okhttp.android.test.test D/okhttp.Http2: << 0x00000000     0 SETTINGS      ACK
2020-04-30 08:19:27.325 27576-27612/okhttp.android.test.test D/okhttp.Http2: << 0x00000003   343 HEADERS       END_HEADERS
2020-04-30 08:19:27.332 27576-27612/okhttp.android.test.test D/okhttp.Http2: << 0x00000003   230 DATA          END_STREAM
2020-04-30 08:19:27.333 27576-27612/okhttp.android.test.test D/okhttp.Http2: << 0x00000000     8 PING          
2020-04-30 08:19:27.337 27576-27615/okhttp.android.test.test D/okhttp.Http2: >> 0x00000000     8 PING          ACK
2020-04-30 08:19:27.539 27576-27609/okhttp.android.test.test D/okhttp.Http2: >> CONNECTION 505249202a20485454502f322e300d0a0d0a534d0d0a0d0a
2020-04-30 08:19:27.540 27576-27609/okhttp.android.test.test D/okhttp.Http2: >> 0x00000000     6 SETTINGS      
2020-04-30 08:19:27.542 27576-27609/okhttp.android.test.test D/okhttp.Http2: >> 0x00000000     4 WINDOW_UPDATE 
2020-04-30 08:19:27.548 27576-27609/okhttp.android.test.test D/okhttp.Http2: >> 0x00000003    49 HEADERS       END_STREAM|END_HEADERS
2020-04-30 08:19:27.602 27576-27613/okhttp.android.test.test D/okhttp.Http2: << 0x00000000    18 SETTINGS      
2020-04-30 08:19:27.605 27576-27613/okhttp.android.test.test D/okhttp.Http2: << 0x00000000     4 WINDOW_UPDATE 
2020-04-30 08:19:27.607 27576-27615/okhttp.android.test.test D/okhttp.Http2: >> 0x00000000     0 SETTINGS      ACK
2020-04-30 08:19:27.607 27576-27613/okhttp.android.test.test D/okhttp.Http2: << 0x00000000     0 SETTINGS      ACK
2020-04-30 08:19:27.612 27576-27613/okhttp.android.test.test D/okhttp.Http2: << 0x00000003   318 HEADERS       END_HEADERS
2020-04-30 08:19:27.618 27576-27613/okhttp.android.test.test D/okhttp.Http2: << 0x00000003  1946 DATA          PADDED
2020-04-30 08:19:27.620 27576-27613/okhttp.android.test.test D/okhttp.Http2: << 0x00000003   248 DATA          END_STREAM|PADDED
2020-04-30 08:19:27.622 27576-27613/okhttp.android.test.test D/okhttp.Http2: << 0x00000000     8 PING          
2020-04-30 08:19:27.625 27576-27618/okhttp.android.test.test D/okhttp.Http2: >> 0x00000000     8 PING          ACK
2020-04-30 08:19:27.742 27576-27609/okhttp.android.test.test I/TestRunner: finished: get(okhttp.android.test.compare.OkHttpClientTest)
2020-04-30 08:19:27.747 27576-27576/okhttp.android.test.test I/MonitoringInstr: Activities that are still in CREATED to STOPPED: 0
2020-04-30 08:19:27.749 27576-27609/okhttp.android.test.test I/TestRunner: run finished: 1 tests, 0 failed, 0 ignored
2020-04-30 08:19:27.930 27576-27576/okhttp.android.test.test I/MonitoringInstr: Activities that are still in CREATED to STOPPED: 0
2020-04-30 08:19:27.930 27576-27609/okhttp.android.test.test I/MonitoringInstr: waitForActivitiesToComplete() took: 0ms
2020-04-30 08:19:27.932 3715-4801/? I/ActivityManager: Force stopping okhttp.android.test.test appid=10060 user=0: finished inst
2020-04-30 08:19:27.932 3715-4801/? I/ActivityManager: Killing 27576:okhttp.android.test.test/u0a60 (adj 0): stop okhttp.android.test.test
2020-04-30 08:19:27.988 3715-27622/? I/ActivityManager: Failure reporting to instrumentation watcher: comp=ComponentInfo{okhttp.android.test.test/androidx.test.runner.AndroidJUnitRunner} results=Bundle[mParcelledData.dataSize=92]
```

File: okhttp/src/main/kotlin/okhttp3/internal/platform/android/AndroidLog.kt [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    this[OkHttpClient::class.java.name] = "okhttp.OkHttpClient"
+    this[Http2::class.java.name] = "okhttp.Http2"
+    this[TaskRunner::class.java.name] = "okhttp.TaskRunner"
+  }.toMap()
+
+  fun enable() {
+    val logger = Logger.getLogger(OkHttpClient::class.java.`package`.name)
+    if (configuredLoggers.add(logger)) {
+      // log everything and filter in androidLog method below
+      logger.level = Level.FINE
```
Comment (yschimke) [inline okhttp/src/main/kotlin/okhttp3/internal/platform/android/AndroidLog.kt:77]: @swankjesse If we are ok with having to call setprop before you run the app, then we can avoid log formatting.  But it just means you can't dynamically adjust after running.

Thoughts?

Comment (yschimke) [reply okhttp/src/main/kotlin/okhttp3/internal/platform/android/AndroidLog.kt:77]: I'll land it with a tweak so you need to restart the app after reloading, we can look at smarter options after that.  But it's really nice with just running

```
$ adb shell setprop log.tag.okhttp.Http2 DEBUG
```

Comment (swankjesse) [reply okhttp/src/main/kotlin/okhttp3/internal/platform/android/AndroidLog.kt:77]: Is it gonna do string formatting on every frame?

Comment (yschimke) [reply okhttp/src/main/kotlin/okhttp3/internal/platform/android/AndroidLog.kt:77]: No. That's exactly what I flagged as I want to avoid.

**Pull request merged**